app/controllers/incomeSources/manage/ConfirmReportingMethodSharedController.scala

```diff
@@ -225,29 +224,40 @@ class ConfirmReportingMethodSharedController @Inject()(val manageIncomeSources:
               journeyType = incomeSourceType.journeyType,
               operationType = "MANAGE",
               reportingMethodChangeTo = formatReportingMethod(reportingMethod),
-              taxYear = taxYears.startYear.toString + "-" + taxYears.endYear.toString,
+              taxYear = taxYear.startYear.toString + "-" + taxYear.endYear.toString,
               businessName = incomeSourceBusinessName.getOrElse("Unknown")
             )
           )
         Future.successful(Redirect(errorCall))
       case res: UpdateIncomeSourceResponseModel =>
-        logAndShowError(isAgent, s"Updated tax year specific reporting method: $res")
-        auditingService
-          .extendedAudit(
-            IncomeSourceReportingMethodAuditModel(
-              isSuccessful = true,
-              journeyType = incomeSourceType.journeyType,
-              operationType = "MANAGE",
-              reportingMethodChangeTo = formatReportingMethod(reportingMethod),
-              taxYear = taxYears.startYear.toString + "-" + taxYears.endYear.toString,
-              businessName = incomeSourceBusinessName.getOrElse("Unknown")
-            )
-          )
-        Future.successful(Redirect(successCall))
-    } recover {
-      case ex: Exception =>
-        logAndShowError(isAgent, s"[handleUpdateSuccess]: Error updating reporting method: ${ex.getMessage}")
+        withSessionData(JourneyType(Manage, incomeSourceType)) {
+          uiJourneySessionData =>
+            val newUIJourneySessionData = {
+              uiJourneySessionData.copy(manageIncomeSourceData =
+                Some(ManageIncomeSourceData(Some(incomeSourceIdMaybe.get.toString), Some(reportingMethod), Some(taxYear.startYear.toString + "-" + taxYear.endYear.toString), Some(true))))
+            }
+            sessionService.setMongoData(newUIJourneySessionData).flatMap {
+              case true => Future.successful(Redirect(successCall))
+              case false => Future.successful(logAndShowError(isAgent, "Unable to update mongo journey as complete"))
```

Comment:
we would never get false from setMongoData; 
might be better to apply next change
def set(data: UIJourneySessionData): Future[Unit]

Reply:
Thanks I'll fix